pkg/core/resources/apis/meshmultizoneservice/api/v1alpha1/meshmultizoneservice.go

```diff
@@ -0,0 +1,42 @@
+// +kubebuilder:object:generate=true
+package v1alpha1
+
+import (
+	hostnamegenerator_api "github.com/kumahq/kuma/pkg/core/resources/apis/hostnamegenerator/api/v1alpha1"
+	meshservice_api "github.com/kumahq/kuma/pkg/core/resources/apis/meshservice/api/v1alpha1"
+)
+
+// MeshMultiZoneService allows users to create a service that spawns across multiple zones
+// It aggregates existing MeshServices by labels.
+// +kuma:policy:is_policy=false
+// +kuma:policy:has_status=true
+type MeshMultiZoneService struct {
+	// Selector is a way to select multiple MeshServices
+	Selector Selector `json:"selector"`
+}
+
+type Selector struct {
+	// MeshService selects MeshServices
+	MeshService MeshServiceSelector `json:"meshService"`
+}
+
+type MeshServiceSelector struct {
+	// MatchLabels matches multiple MeshServices by labels
+	MatchLabels map[string]string `json:"matchLabels"`
+}
+
+type MeshMultiZoneServiceStatus struct {
+	// Ports is an aggregated distinct list of ports from selected MeshServices
+	Ports []meshservice_api.Port `json:"ports,omitempty"`
```

Comment:
How useful will aggregated information be? I can imagine a situation where you can see a few ports, but in one zone, there is only one. In this case, it might be confusing for me. Do we need information indicating which zone/meshservice is the origin of this entry, or do we want to base it on the MeshServices list?

Reply:
> I can imagine a situation where you can see a few ports, but in one zone, there is only one

What would be that scenario in practice?

I think in 99,9% cases you aggregate the same service over different clusters.

Reply:
If you integrate K8s and Universal, it could happen. But true, most of the time they should aggregate the same service